creator-node/src/services/stateMachineManager/stateReconciliation/recoverOrphanedData.jobProcessor.ts

```diff
@@ -142,67 +137,146 @@ const _storeWalletsWithThisNodeInReplica = async (
   const walletsWithNodeInReplicaSetArr: string[] = []
   do {
     // Get batch of users and all it to the array of all wallets
-    batchOfUsers = await getNodeUsers(
-      discoveryNodeEndpoint,
-      thisContentNodeEndpoint,
-      prevUserId,
-      NUM_USERS_PER_QUERY
-    )
-    batchOfUsers?.forEach((user) =>
-      walletsWithNodeInReplicaSetArr.push(user.wallet)
-    )
+    try {
+      batchOfUsers = await getNodeUsers(
+        discoveryNodeEndpoint,
+        thisContentNodeEndpoint,
+        prevUserId,
+        NUM_USERS_PER_QUERY
+      )
+      batchOfUsers?.forEach((user) =>
+        walletsWithNodeInReplicaSetArr.push(user.wallet)
+      )
 
-    // Move pagination cursor to the end of the batch
-    prevUserId = batchOfUsers?.length
-      ? batchOfUsers[batchOfUsers.length - 1].user_id
-      : 0
+      // Move pagination cursor to the end of the batch
+      prevUserId = batchOfUsers?.length
+        ? batchOfUsers[batchOfUsers.length - 1].user_id
+        : 0
+    } catch (e: any) {
+      logger.error(
+        `Error fetching batch of users from ${discoveryNodeEndpoint}: ${e.message}`
+      )
+      // `batchOfUsers?.length` will be 0, which will cause us to break out of this loop and not fetch all users.
+      // This will cause extra users to be marked as orphaned, which is okay because orphaned data
+      // recovery will short circuit later to avoid wiping state on any node in the user's replica set.
+    }
```

Comment:
worth calling out here this is after retries
also, another reason to use for loop 🙂